nixos/modules/profiles/macos-builder.nix

```diff
@@ -93,7 +96,7 @@ in
         };
       });
 
-  system.stateVersion = "22.05";
+  inherit (config.system) stateVersion;
```

Comment:
There are two issues here:

- This sets `stateVersion` and not `system.stateVersion`
- If it were changed to set `system.stateVersion` then this also seems like it would also trigger infinite recursion.